source/AI.cpp

```diff
@@ -3296,10 +3295,15 @@ void AI::MovePlayer(Ship &ship, const PlayerInfo &player, Command &activeCommand
 						continue;
 					
 					bool isEnemy = other->GetGovernment()->IsEnemy(ship.GetGovernment());
-					double d = other->Position().Distance(ship.Position());
-					if((isEnemy && !foundEnemy) || (d < closest && isEnemy == foundEnemy))
+					double b;
+					if(Preferences::Has("Boarding target priority")) {
+						b = other->Position().Distance(ship.Position());
+					} else {
+						b = -other->Cost();
+					}
```

Comment:
You can replace this with a ternary operator, it looks better imo:

```c++
double b = distancePriority ? other->Position().Distance(ship.Position()) : -other->Cost();
```

Reply:
you're right it's more clear with a bool signalling it represents distancePriority or not, implemented both requested changes